### PR TITLE
Use pytest fixtures for anndata tests

### PR DESCRIPTION
**Pull request by RK900**
referenced in #1574

### CODE REVIEW

File: tests/dataset/test_anndata.py [hunk 1]
```diff
@@ -329,25 +310,22 @@ def test_setup_anndat_create_label_batch():
...
     adata.obs["batch"][:10] = np.nan
     with pytest.raises(ValueError):
         generic_setup_adata_manager(adata, batch_key="batch")
 
 
-def test_setup_anndata_cat():
+def test_setup_anndata_cat(adata):
     # test error is thrown when categorical joint obsm field contains nans
-    adata = synthetic_iid()
     adata.obs["cat1"] = np.random.randint(0, 5, size=(adata.shape[0],))
```
Comment (jjhong922) [inline tests/dataset/test_anndata.py:322]: this stuff for example can be factored out

File: tests/dataset/test_anndata.py [hunk 2]
```diff
@@ -489,8 +462,7 @@ def test_anntorchdataset_getitem_pro_exp():
         assert type(value) == np.ndarray
 
 
-def test_view_registry():
-    adata = synthetic_iid()
+def test_view_registry(adata):
     adata.obs["cont1"] = np.random.normal(size=(adata.shape[0],))
```
Comment (jjhong922) [inline tests/dataset/test_anndata.py:466]: same here, wont comment for rest of the file

**Review by jjhong922 — COMMENTED** [review]
thanks! Left some comments, look for more places to extract code. Remember the fixture can contain more setup than currently exists in every tests. We want each fixture to cover as much of the setup for 90% of tests that need them.

**Comment by RK900** [comment]
I tried factoring out the model training into a fixture, but that resulted in test_anndata taking 4->14 seconds to run

File: tests/dataset/conftest.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import numpy as np
+import pytest
+
```
Comment (jjhong922) [inline tests/dataset/conftest.py:3]: we will want to use this fixture in other test files like test_models.py for example, so probably want to put it in a higher level conftest

**Review by jjhong922 — APPROVED** [review]
looks good to me, you can ignore the comment for now and move it up when you make the changes to other test files.

**Comment by jjhong922** [comment]
> I tried factoring out the model training into a fixture, but that resulted in test_anndata taking 4->14 seconds to run

did you use the fixture on every test, or was it only used for tests that did this anyways? If its the latter I dont understand why the runtime would increase so much.

**Comment by RK900** [comment]
It was only for the tests that required a model to be trained.

**Pull request merged**